web/functions.php

```diff
@@ -53,6 +54,18 @@ function getShieldURL($query, $defaults): string
         "key" => $key
     ));
 
+    //getting the data from the url as JSON
+    $response = curl_get_contents($apiUrl);
+
+    // decoding the json data hence we can extract the Subscribers count
+    $decoded = json_decode($response);
+
+    // getting the subscribers count by querying throught the decoded json
+    $number = $decoded->items[0]->statistics->$query;
+
+    // getting the subscribers count to a rounded value and right format
+    $message = shortNumber($number);
+
```

Comment:
At this point a function that formats the message based on the `format` parameter should be called. The `formatResponseNumber` can be simplified greatly since the new method of getting the data is much simpler.

The regex matching is not necessary and something that just takes in the number, style, and format and returns the number formatted according to the case would be enough (`number_format($number)` for "commas", `shortNumber($number)` for "short" but keep the `if ('for-the-badge' === $style)`, for "none", it should just return the number.

To fix the issue, the function should be called at this point, before generating the shields URL.

Reply:
Hey @DenverCoder1, I made the required changes in the functions.php file, please review them and inform me of any changes.